src/emulator/commandUtils.ts

```diff
@@ -464,9 +465,11 @@ export async function emulatorExec(script: string, options: any): Promise<void>
   try {
     const showUI = !!options.ui;
     ({ deprecationNotices } = await controller.startAll(options, showUI, true));
+    sendVSCodeMessage({ message: VSCODE_MESSAGE.EMULATORS_STARTED });
     exitCode = await runScript(script, extraEnv);
     await controller.onExit(options);
   } finally {
+    sendVSCodeMessage({ message: VSCODE_MESSAGE.EMULATORS_START_ERRORED });
```

Comment:
Wouldn't having this in the `finally` block _always_ notify the emulator as errored?

Reply:
+1 - this should be in a new catch, and there should be a separate finally block for controller.cleanShutdown()

Reply:
Oh I just didn't read that at all. good finally, I mean catch